bpm/bonita-core/bonita-login/bonita-login-api-impl/src/main/java/org/bonitasoft/engine/core/login/SecuredLoginServiceImpl.java

```diff
@@ -112,7 +112,8 @@ public SSession login(final Map<String, Serializable> credentials) throws SLogin
         } catch (final AuthenticationException ae) {
             throw new SLoginException(ae);
         } catch (final SUserNotFoundException e) {
-            throw new SLoginException("Unable to found user " + userName);
+            warningLog("Unable to find user with username " + userName + " in database.");
+            throw new SLoginException("Unable to find user " + userName);
         } finally {
```

Comment:
The problem is that this is the same exception type that is thrown (SLoginException) whether the credentials are invalid (first catch) or the user is not found because of an installation issue (second catch) and we have no way to differentiate client-side (we don't want to log for each LoginException...). And we cannot change the signature of the login method to throw a UserNotFoundException (API break).
Regarding the fact that it is logged in warning, it seemed more appropriate to me than debug since it means there is an installation issue (the users in bonita BD haven't been synchronised with the external referential (see BS-9604)).